XenAdmin/TabPages/PhysicalStoragePage.cs

```diff
@@ -337,6 +336,41 @@ private void dataGridViewSr_SortCompare(object sender, DataGridViewSortCompareEv
                 e.SortResult = diff > 0 ? 1 : diff < 0 ? -1 : 0;
                 e.Handled = true;
             }
+            else if (e.Column.Index == columnName.Index)
+            {
+                sr1CompStr = sr1.Name();
+                sr2CompStr = sr2.Name();
+            }
+            else if (e.Column.Index == columnDescription.Index)
+            {
+                sr1CompStr = sr1.Description();
+                sr2CompStr = sr2.Description();
+            }
+            else if (e.Column.Index == columnType.Index)
+            {
+                sr1CompStr = sr1.FriendlyTypeName();
+                sr2CompStr = sr2.FriendlyTypeName();
+            }
+            else if (e.Column.Index == columnShared.Index)
+            {
+                sr1CompStr = sr1.shared ? Messages.YES : Messages.NO;
+                sr2CompStr = sr2.shared ? Messages.YES : Messages.NO;
+            }
+
+            if (sr1CompStr != null && sr2CompStr != null)

```

Comment:
Instead of relying on the developer setting or not setting this in the previous if blocks, it would be safer if you returned after setting e.Handled in the previous blocks.